strr-examiner-web/app/stores/examiner.ts

```diff
@@ -40,7 +40,54 @@ export const useExaminerStore = defineStore('strr/examiner-store', () => {
   const nocContent = reactive({
     content: ''
   })
-  // TODO: Add assigned logic check later
+  const isEditingRentalUnit = ref(false)
+  const hasUnsavedRentalUnitChanges = ref(false)
+  const rentalUnitAddressToEdit = ref<Partial<EditStrAddress>>({})
+  const rentalUnitAddressSchema = computed(() => z.object({
+    addressLineTwo: z.string().optional().default(''),
+    city: z.string().min(1, t('validation.address.city')),
+    province: z.string().min(1, t('validation.address.region')),
+    postalCode: z.string().min(1, t('validation.address.postalCode')),
+    locationDescription: z.string().optional().default(''),
+    streetName: z.string().optional().default(''),
+    streetNumber: z.string().optional().default(''),
+    unitNumber: z.string().optional().default('')
+  }).superRefine((data, ctx) => {
+    if (data.addressLineTwo === '') {
+      if (!data.streetName) {
+        ctx.addIssue({
+          code: 'custom',
+          path: ['streetName'],
+          message: t('validation.address.streetName')
+        })
+      }
+      if (!data.streetNumber) {
+        ctx.addIssue({
+          code: 'custom',
+          path: ['streetNumber'],
+          message: t('validation.address.streetNumber')
+        })
+      }
+    }
+  }))
+  const startEditRentalUnitAddress = () => {
+    const addressData = JSON.parse(JSON.stringify(activeReg.value?.unitAddress || {}))
+    if (addressData.locationDescription === null) {
+      addressData.locationDescription = ''
+    }
+    rentalUnitAddressToEdit.value = addressData
+    isEditingRentalUnit.value = true
+    hasUnsavedRentalUnitChanges.value = false // Reset unsaved changes flag
+  }
+  const resetEditRentalUnitAddress = () => {
+    isEditingRentalUnit.value = false
+    rentalUnitAddressToEdit.value = {}
+    hasUnsavedRentalUnitChanges.value = false
+  }
+  const setHasUnsavedRentalUnitChanges = (value: boolean) => {
```

Comment:
`setHasUnsavedRentalUnitChanges` is redundant, because `hasUnsavedRentalUnitChanges` is already exposed from store as ref and can be updated directly (as on line 80).

Reply:
Done